cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -167,6 +168,12 @@ struct ARROW_EXPORT S3Options {
   /// delay between retries.
   std::shared_ptr<S3RetryStrategy> retry_strategy;
 
+  /// Coalesce options for requesting multiple reads at once
+  io::CoalesceOptions coalesceOptions = io::CoalesceOptions::Defaults();
```

Comment:
nit: follow the naming conventions (snake_case)

Reply:
thanks, btw I'll use here `io::CoalesceOptions::MakeFromNetworkMetrics(5, 500)` as default value (i.e. `hole_size_limit = 2.5 MiB` and `range_size_limit = 22.5 MiB`)